### PR TITLE
Task 6 Work with HTTP

### PR DESCRIPTION
**Pull request by Laranto-spb**
Problems with PropTypes

### CODE REVIEW

File: src/components/card/card-dialog/CardDialog.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ function CardDialog({
               label="Card title"
               variant="outlined"
               onChange={handleTitleChange}
-              value={cardTitle}
+              value={cardTitle || ''}
```
Comment (Laranto-spb) [inline src/components/card/card-dialog/CardDialog.js:75]: Fixed null value error when card is new.

File: src/components/card/card-dialog/CardDialog.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ CardDialog.propTypes = {
   addCard: PropTypes.func,
   updCard: PropTypes.func,
   card: PropTypes.shape({
-    id: PropTypes.string,
+    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (Laranto-spb) [inline src/components/card/card-dialog/CardDialog.js:113]: Does not work, still wants id as a string (number comes from api)

File: src/components/main/Main.js [hunk 3]
```diff
@@ -1,33 +1,26 @@
...
 function Main() {
-  const [cards, setCards] = useState(CARDS);
+  const [cards, setCards] = useState([]);
+
+  useEffect(() => {
+    axios.get('https://jsonplaceholder.typicode.com/posts')
+      .then((response) => {
+        setCards(response.data);
+      });
+  }, []);
```
Comment (Laranto-spb) [inline src/components/main/Main.js]: Should I re-request when cards are changed - if we have not a dummy-server? 
Add new card -> card variable changed -> post request -> get request (useEffect)

File: src/components/card/card-dialog/CardDialog.js (hunk 2 above)
Comment (Artsiom-L) [reply src/components/card/card-dialog/CardDialog.js:113]: You forgot to add those types to CardItem component

File: src/components/main/Main.js (hunk 3 above)
Comment (Artsiom-L) [reply src/components/main/Main.js]: No, we need that only for initialization

File: src/components/main/Main.js [hunk 4]
```diff
@@ -1,33 +1,26 @@
...
-    body: 'Text of Card 3',
-  },
-];
-
 function Main() {
-  const [cards, setCards] = useState(CARDS);
+  const [cards, setCards] = useState([]);
+
+  useEffect(() => {
+    axios.get('https://jsonplaceholder.typicode.com/posts')
```
Comment (Artsiom-L) [inline src/components/main/Main.js]: Could you create an API client for those stuff ?

File: src/components/main/Main.js [hunk 5]
```diff
@@ -1,33 +1,26 @@
...
+      .then((response) => {
+        setCards(response.data);
+      });
+  }, []);
 
   const setNewCard = (newCard) => {
+    axios.post('https://jsonplaceholder.typicode.com/posts')
```
Comment (Artsiom-L) [inline src/components/main/Main.js]: The same thing about an API client

File: src/components/card/card-dialog/CardDialog.js (hunk 2 above)
Comment (Laranto-spb) [reply src/components/card/card-dialog/CardDialog.js:113]: Exactly!

File: src/components/main/Main.js [hunk 6]
```diff
@@ -1,26 +1,22 @@
...
   const [cards, setCards] = useState([]);
 
   useEffect(() => {
-    axios.get('https://jsonplaceholder.typicode.com/posts')
-      .then((response) => {
-        setCards(response.data);
-      });
+    cardApiResult.cards.getAll()
+      .then((res) => setCards((res.data)));
   }, []);
```
Comment (Laranto-spb) [inline src/components/main/Main.js]: I get a promise from api client

**Review by Artsiom-L — APPROVED** [review]

**Pull request merged**